services/user/Supervisor/ui/src/oauth/App.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect } from "react";
+import { siblingUrl } from "@psibase/common-lib";
+import { ActiveOauthRequest, isTypeOauthRequest, OauthRequest } from "./db";
+
+const buildIframeUrl = (promptUserReq: OauthRequest) => {
+    const urlParams = new URLSearchParams(window.location.search);
+    const url = siblingUrl(
+        null,
+        promptUserReq.subdomain,
+        promptUserReq.subpath,
+        true,
+    );
+    const newIframeUrl = new URL(url);
+    newIframeUrl.searchParams.set("id", promptUserReq.id);
+    newIframeUrl.searchParams.set(
+        "returnUrlPath",
+        urlParams.get("returnUrlPath") || "",
+    );
+    return newIframeUrl.toString();
+};
+
+export const App = () => {
+    const [iframeUrl, setIframeUrl] = useState<string | null>(null);
+    const [error, setError] = useState<string | null>(null);
+
+    const initApp = async () => {
+        const urlParams = new URLSearchParams(window.location.search);
+        const oauth_id = urlParams.get("id");
+        if (!oauth_id) {
+            setError("OAuth request error: No id provided");
+            return;
+        }
+
+        try {
+            const oauthReq = await ActiveOauthRequest.get(oauth_id);
+            if (isTypeOauthRequest(oauthReq)) {
+                setIframeUrl(buildIframeUrl(oauthReq as OauthRequest));
+                ActiveOauthRequest.delete();
+            } else {
+                setError("OAuth request error: Invalid payload");
+            }
+        } catch (e) {
+            setError("OAuth request error: " + e);
+        }
+    };
+
+    useEffect(() => {
+        initApp();
+    }, []);
+
+    if (!!error) {
+        return <div>{error}</div>;
+    }
+
+    return (
+        <div
+            style={{
+                width: "100%",
+                height: "100vh",
+                display: "flex",
+                justifyContent: "center",
+                alignItems: "center",
+            }}
+        >
+            {iframeUrl && (
+                <iframe
+                    src={iframeUrl}
+                    style={{
+                        width: "50%",
+                        height: "50vh",
+                        border: "none",
+                    }}
+                />
+            )}
+        </div>
```

Comment:
I recommend using `dvh` units here instead of `vh` units, in order to better support mobile browsers with dynamic viewport heights.